src/lib/RequestParser.ts

```diff
@@ -148,28 +152,75 @@ export class RequestParser {
-                    console.warn(
-                        'CashlinkRequest.cashlinkAddress and CashlinkRequest.senderAddress should not both be defined.',
-                    );
+            case RequestType.CREATE_CASHLINK:
+                const createCashlinkRequest = request as CreateCashlinkRequest;
+                const senderAddress = 'senderAddress' in createCashlinkRequest
+                    ? Nimiq.Address.fromUserFriendlyAddress(createCashlinkRequest.senderAddress)
+                    : undefined;
+                const senderBalance = 'senderBalance' in createCashlinkRequest
+                    ? createCashlinkRequest.senderBalance
+                    : undefined;
+                if (senderBalance !== undefined && !Nimiq.NumberUtils.isUint64(senderBalance)) {
+                    throw new Error('Invalid Cashlink senderBalance');
+                }
+
+                const value = createCashlinkRequest.value;
+                if (value !== undefined && (!Nimiq.NumberUtils.isUint64(value) || value === 0)) {
+                    throw new Error('Malformed Cashlink value');
+                }
+
+                let message = 'message' in createCashlinkRequest ? createCashlinkRequest.message : undefined;
+                if (message !== undefined) {
+                    if (typeof message !== 'string') {
+                        throw new Error('Cashlink message must be a string');
+                    }
+                    const { result: truncated, didTruncate } = Utf8Tools.truncateToUtf8ByteLength(message, 255);
+                    if (!('autoTruncateMessage' in createCashlinkRequest && createCashlinkRequest.autoTruncateMessage)
+                        && didTruncate) {
+                        throw new Error('Cashlink must be shorter than 256 bytes or autoTruncateMessage must be '
+                            + 'enabled.');
+                    }
+                    message = truncated;
+                }
+
+                const theme = createCashlinkRequest.theme || CashlinkTheme.UNSPECIFIED;
+                if (!Object.values(CashlinkTheme).includes(theme) || !Nimiq.NumberUtils.isUint8(theme)) {
+                    // Also checking whether theme is a valid Uint8 to catch ids that are potentially to high in the
+                    // CashlinkTheme enum and to filter out values that are actually the enum keys that have been added
+                    // by typescript for reverse mapping.
+                    throw new Error('Invalid Cashlink theme');
                 }
+
+                const returnCashlink = !!createCashlinkRequest.returnCashlink;
+                if (returnCashlink && !isPriviledgedOrigin(state.origin)) {
+                    throw new Error(`Origin ${state.origin} is not authorized to request returnCashlink.`);
+                }
+                const skipSharing = !!createCashlinkRequest.returnCashlink && !!createCashlinkRequest.skipSharing;
```

Comment:
Only a suggestion.

```suggestion
                const skipSharing = returnCashlink && !!createCashlinkRequest.skipSharing;
```

Reply:
It's actually not possible to do that simplification here, because `createCashlinkRequest.skipSharing` only exists if `createCashlinkRequest.returnCashlink` is defined.
Unfortunately ts does not recognize that `returnCashlink` is the same as `createCashlinkRequest.returnCashlink`.